lib/daru/date_time/index.rb

```diff
@@ -366,6 +366,24 @@ def [] *key
       end
     end
 
+    def pos *args
+      # to filled
+      out = self[*args]
+      return out if out.is_a? Numeric
+      out.map { |date| self[date] }
+    end
+
+    def subset *args
+      self[*args]
+    end
+
+    def respond? *args
```

Comment:
@lokeshh your documentation for the method in index.rb says "Returns true if all arguments are either a valid category or position". So, I think `valid?` would be the most reasonable. At least, looking at `if @index.valid?(:foo)`, the reader can do some reasonable assumption on it.